kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java

```diff
@@ -461,22 +494,22 @@ private void collectFormData() {
     public void onDone(ActionEvent actionEvent) {
         actionEvent.consume();
         collectFormData();
-
         // save calls validate
         patternFieldsViewModel.save();
         if (patternFieldsViewModel.hasErrorMsgs()) {
             // when there are validators, we potentially will have errors
             return; // do not proceed.
         }
-
         //publish close env
         EvtBusFactory.getDefaultEvtBus().publish(patternFieldsViewModel.getPropertyValue(PATTERN_TOPIC),
                 new PatternPropertyPanelEvent(actionEvent.getSource(), CLOSE_PANEL));
 
-
+        //TODO This If block is not required but for some reason the bidirectional binding declared in initialize method is not working with FIELD_ORDER.
+        if(fieldOrderComboBox.getSelectionModel().getSelectedItem() != patternFieldsViewModel.getValue(FIELD_ORDER)){
```

Comment:
Something is not wired up correctly. 
```java
// The code below should be in the initialize() method
ObjectProperty<Integer> fieldOrderProp = patternFieldsViewModel.getProperty(FIELD_ORDER);
fieldOrderProp.bind(fieldOrderComboBox.getSelectionModel().selectedItemProperty()); 
```
If you'll notice this is not bidirectional. The user is selecting an item that will sync.